PRACTICE.md

```diff
@@ -157,5 +157,52 @@ Keep close track of both of these processes so that you can answer any questions
+
+Software testing can be conducted as soon as executable software (even if partially complete) exists. The overall approach to software development often determines when and how testing is conducted. For example, in a phased process, most testing occurs after system requirements have been defined and then implemented in testable programs.
+
+In the MASQ Network project, the process of QA is triggered when a card is moved into the 'Quality Assurance In Progress' column on our [Card Wall](https://github.com/MASQ-Project/Node/blob/master/COLUMNS.md)
+
+The Testing Supervisor or QA Manager is responsible for passing or failing a card based on the feedback from the test team across the different operating systems being supported. Any feedback, bugs or suggestions are communicated to the developer of the card, and another iteration of development, testing and review is completed.
+
+CLI-based QA requires some basic command-line knowledge and understanding of the Node software itself.
+
+If you would like to contribute to QA testing, and have a good foundational knowledge of CLI, please reach out to our team by [email](mailto:info@masq.ai) or join our [Discord](https://discord.gg/masq) and tag one of our admins.
+
+## Further Information on Versioned Releases
+The determination of versioning will start with a discussion of the core developers and the Product Owner. This will begin towards the end of a cards engineering practice above, most likely during final review and QA steps.
+
+Versioning follows basic [semver format](https://semver.org) - view the formal documentation and conventions for reference.
+
+The core specifications to be followed are:
+
+- Once a versioned package has been released, the contents of that version MUST NOT be modified. Any modifications MUST be released as a new version.
+- A pre-release version MAY be denoted by appending a hyphen and a series of dot separated identifiers immediately following the patch version. Identifiers MUST comprise only ASCII alphanumerics and hyphens [0-9A-Za-z-]. Identifiers MUST NOT be empty. Numeric identifiers MUST NOT include leading zeroes. Pre-release versions have a lower precedence than the associated normal version. A pre-release version indicates that the version is unstable and might not satisfy the intended compatibility requirements as denoted by its associated normal version. Examples: 1.0.0-alpha, 1.0.0-alpha.1, 1.0.0-0.3.7, 1.0.0-x.7.z.92, 1.0.0-x-y-z.–.
+
+### Steps Followed for Version Increment
```

Comment:
Do we want to have non-release candidates tagged or just leave them as normal merges into master? Ex 3 cards merge into master, not one of the cards warrants a release by itself. Once the last card is going through PR can the cargo.tomls be incremented for that branch and tag the commit for release.
This will give a cleaner repo vs having dozens of pre-release.x tags, while also simplifying QA within React-app using "latest" binaries. If we want to start using versioned workflows within React-app this will require non-trivial testing.

Reply:
Well I guess the main reason for the tagging of non-release candidates is that its changing the master codebase each time.
so lets say something does need to be identified or rolled back (or even tested) then its easy to checkout the git tag of the master branch because every merge had a git tag, instead of searching based on commits and commit SHA.

But to be clear I didn't intend on the cargo.tomls being edited each time with the pre-release.x versions, its only for the git tag.

But to be honest this can be managed either way with clear and concise communication within the devs team and whoever is conducting the merges. Since master branch is protected it could be enforced by maintainers, which current are Dan, Bert and I.
before any merge this could be checked

Reply:
bert had a good conversation on this also @FinsaasGH 

we will forgo git tagging every commit, and only the commit to master that will have a release candidate.
if it needs to be a prerelease at that time, we can git tag it and have the release package labelled prerelease for testing if needed